manifests/init.pp

```diff
@@ -343,4 +343,10 @@
 
   contain 'foreman::settings' # lint:ignore:relative_classname_inclusion (PUP-1597)
   Class['foreman::database'] -> Class['foreman::settings']
+
+  file { '/usr/share/foreman/tmp/restart_required_changed_plugins':
+    ensure  => absent,
+    notify  => Class['foreman::service'],
```

Comment:
I recommend adding rspec coverage for this.  Notifications seem to frequently become accidentally broken during a refactor.

Reply:
You're right!

Updated.